ERCS/erc-7704.md

```diff
@@ -0,0 +1,172 @@
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract SRT20Minimal is ERC20 {
+    error TransferNotAllowed();
+
+    constructor() ERC20("Soul Resonance Token 20", "SRT20") {}
+
+    function _update(
+        address from,
+        address to,
+        uint256 value
+    ) internal virtual override {
+        if (from != address(0) && to != address(0)) {
+            revert TransferNotAllowed();
+        }
+        super._update(from, to, value);
+    }
+
+    function mint(address to, uint256 amount) public {
+        _mint(to, amount);
+    }
+
+    function burn(address from, uint256 amount) public {
+        _burn(from, amount);
+    }
+
+}
+```
+
+```solidity
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+contract SRT721Minimal is ERC721 {
+    error TransferNotAllowed();
+
+    constructor() ERC721("Soul Resonance Token 721", "SRT721") {}
+
+    function _update(
+        address to,
+        uint256 tokenId,
+        address auth
+    ) internal virtual override returns (address) {
+        address from = super._update(to, tokenId, auth);
+        if (from != address(0) && to != address(0)) {
+            revert TransferNotAllowed();
+        }
+        return from;
+    }
+
+    function safeMint(address to, uint256 tokenId) public {
+        _mint(to, tokenId);
+    }
+
+    function burn(uint256 tokenId) public {
+        _update(address(0), tokenId, msg.sender);
+    }
+
+}
+```
+
+```solidity
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+
+contract SRT1155Minimal is ERC1155 {
+    error TransferNotAllowed();
+
+    constructor() ERC1155("srt") {}
+
+    function _update(
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory values
+    ) internal virtual override {
+        if (from != address(0) && to != address(0)) {
+            revert TransferNotAllowed();
+        }
+        super._update(from, to, ids, values);
+    }
+
+    function mint(
+        address to,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
+    ) public {
+        _mint(to, id, amount, data);
+    }
+
+    function burn(
+        address account,
+        uint256 id,
+        uint256 amount
+    ) public {
+        _burn(account, id, amount);
+    }
+}
+```
+
+## Security Considerations
+
+While we maintain high standards for code quality and test coverage throughout the development process of this project, it is crucial to acknowledge that the overall security of the implementation cannot be guaranteed at this time. The Soul Resonance Token (SRT) protocol and its associated smart contracts have not yet undergone a comprehensive, independent security audit by a reputable third-party organization.
```

Comment:
You should write as if your proposal is already Final (meaning immutable.) Once this document goes to Final, you won't be able to update this text, then if you ever do get an audit, this section will be incorrect.